File: 1108-defanging-an-ip-address/1108-defanging-an-ip-address.c


char * defangIPaddr(char * address){

    char *d = calloc(1,sizeof("xxx[.]xxx[i]xxx[i]xxx"));
    char *t = d;
    for (char *a = address; *a ; a++){
        *a == '.' ? *d++ = '[', *d++ = '.' , *d++ = ']' : (*d++ = *a);
    }
    return t;
}

File: 1108-defanging-an-ip-address/1108-defanging-an-ip-address.java
class Solution {
    public String defangIPaddr(String address) {
        String word = "";
        for (int i = 0; i < address.length();i++){
            if (address.charAt(i) == '.'){
                word += "[.]";
            }else{
                word += address.charAt(i);
            }
        }
        return word;   
    }
}